Author: Andreas Beckmann <anbe@debian.org>
Description: ignore __GNUC_MINOR__ from GCC 5 onwards
 starting with GCC 5, the major version will be bumped for each release
 and __GNUC_MINOR__ will only denote bugfix releases

--- a/conftest.sh
+++ b/conftest.sh
@@ -1747,7 +1747,7 @@ case "$6" in
         kernel_cc_minor=`echo ${kernel_cc_version} | cut -d '.' -f 2`
 
         echo "
-        #if (__GNUC__ != ${kernel_cc_major}) || (__GNUC_MINOR__ != ${kernel_cc_minor})
+        #if (__GNUC__ != ${kernel_cc_major}) || ((__GNUC__ < 5) && (__GNUC_MINOR__ != ${kernel_cc_minor}))
         #error \"cc version mismatch\"
         #endif
         " > conftest$$.c
